build.sh

```diff
@@ -69,6 +69,19 @@ initDistroRid()
     fi
 }
 
+initTargetDistroRid()
```

Comment:
Instead of this function, why can we not fix https://github.com/dotnet/coreclr/pull/8421/files#diff-0b83f9dedf40d7356e5ca147a077acb4R67 to use `$__BuildArch` instead of `$__HostArch`?